From 750813055e6e4546232fa3020cb1f58fdee86bb5 Mon Sep 17 00:00:00 2001
From: wang-bin <wbsecg1@gmail.com>
Date: Fri, 5 Jun 2020 14:05:42 +0800
Subject: [PATCH 18/30] nvenc: check runtime 9.1 function ptrs before use

---
 libavcodec/nvenc.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/libavcodec/nvenc.c b/libavcodec/nvenc.c
index 76a58bde36..e2e4db52e3 100644
--- a/libavcodec/nvenc.c
+++ b/libavcodec/nvenc.c
@@ -147,7 +147,7 @@ static int nvenc_print_error(AVCodecContext *avctx, NVENCSTATUS err,
     NvencContext *ctx = avctx->priv_data;
     NV_ENCODE_API_FUNCTION_LIST *p_nvenc = &ctx->nvenc_dload_funcs.nvenc_funcs;
 
-    if (p_nvenc && ctx->nvencoder)
+    if (p_nvenc && ctx->nvencoder && p_nvenc->nvEncGetLastErrorString)
         details = p_nvenc->nvEncGetLastErrorString(ctx->nvencoder);
 #endif
 
@@ -1490,7 +1490,7 @@ static av_cold int nvenc_setup_encoder(AVCodecContext *avctx)
     }
 
 #ifdef NVENC_HAVE_CUSTREAM_PTR
-    if (ctx->cu_context) {
+    if (ctx->cu_context && p_nvenc->nvEncSetIOCudaStreams) {
         nv_status = p_nvenc->nvEncSetIOCudaStreams(ctx->nvencoder, &ctx->cu_stream, &ctx->cu_stream);
         if (nv_status != NV_ENC_SUCCESS) {
             nvenc_pop_context(avctx);
-- 
2.37.1 (Apple Git-137.1)

