From 8bef8d1a7a1dd619f91795a01e86f1c4a314f3bf Mon Sep 17 00:00:00 2001
From: wang-bin <wbsecg1@gmail.com>
Date: Fri, 30 Mar 2018 10:46:43 +0800
Subject: [PATCH 12/34] configure: do not filter -libpath for msvc

---
 configure | 1 +
 1 file changed, 1 insertion(+)

diff --git a/configure b/configure
index 4fbb2827c5..c38632adb1 100755
--- a/configure
+++ b/configure
@@ -4846,6 +4846,7 @@ msvc_common_flags(){
             -lz)                  echo zlib.lib ;;
             -lx264)               echo libx264.lib ;;
             -lstdc++)             ;;
+            -libpath:*|-LIBPATH:*)echo $flag ;;
             -l*)                  echo ${flag#-l}.lib ;;
             -LARGEADDRESSAWARE)   echo $flag ;;
             -L*)                  echo -libpath:${flag#-L} ;;
-- 
2.39.5 (Apple Git-154)

