From da0bff2cccb9e6e9017bc4f1bba9076d2d21fc11 Mon Sep 17 00:00:00 2001
From: wang-bin <wbsecg1@gmail.com>
Date: Tue, 11 Sep 2018 23:16:49 +0800
Subject: [PATCH 13/34] configure: enable icf

---
 configure | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/configure b/configure
index c38632adb1..894142edaa 100755
--- a/configure
+++ b/configure
@@ -7556,6 +7556,9 @@ test_cc -mno-red-zone <<EOF && noredzone_flags="-mno-red-zone"
 int x;
 EOF
 
+check_ldflags --icf=safe || check_ldflags -Wl,--icf=safe || { # ld or clang7 + lld
+    check_cflags -faddrsig && (check_ldflags --icf=all || check_ldflags -Wl,--icf=all)
+}
 
 if enabled icc; then
     # Just warnings, no remarks
-- 
2.39.5 (Apple Git-154)

