From 0629148092bc892281652387254bf358c5c9e4d4 Mon Sep 17 00:00:00 2001
From: wang-bin <wbsecg1@gmail.com>
Date: Tue, 8 Dec 2015 11:15:20 +0800
Subject: [PATCH 1/2] ignore -fPIC for armasm

---
 configure | 1 +
 1 file changed, 1 insertion(+)

diff --git a/configure b/configure
index 3d1ee49..7bb4be9 100755
--- a/configure
+++ b/configure
@@ -3492,6 +3492,7 @@ armasm_flags(){
         case $flag in
             # Filter out MSVC cl.exe options from cflags that shouldn't
             # be passed to gas-preprocessor
+            -fPIC)                                              ;;
             -M[TD]*)                                            ;;
             *)                  echo $flag                      ;;
         esac
-- 
2.6.3.windows.1

