From 9bc3c279effc3606ab6778eda870228c31e64b18 Mon Sep 17 00:00:00 2001
From: wang-bin <wbsecg1@gmail.com>
Date: Tue, 11 Sep 2018 23:16:49 +0800
Subject: [PATCH 13/36] configure: enable icf

---
 configure | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/configure b/configure
index e7ca030996..2dca7b7066 100755
--- a/configure
+++ b/configure
@@ -8012,6 +8012,9 @@ test_cc -mno-red-zone <<EOF && noredzone_flags="-mno-red-zone"
 int x;
 EOF
 
+check_ldflags --icf=safe || check_ldflags -Wl,--icf=safe || { # ld or clang7 + lld
+    check_cflags -faddrsig && (check_ldflags --icf=all || check_ldflags -Wl,--icf=all)
+}
 
 if enabled icc; then
     # Just warnings, no remarks
-- 
2.50.1 (Apple Git-155)

