From 17a6ab16b765fff6099b1baf5c246f1309de1024 Mon Sep 17 00:00:00 2001
From: wang-bin <wbsecg1@gmail.com>
Date: Tue, 19 Nov 2024 19:43:02 +0800
Subject: [PATCH 31/32] hwcontext_vaapi: fix undefined close() on windows

---
 libavutil/hwcontext_vaapi.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/libavutil/hwcontext_vaapi.c b/libavutil/hwcontext_vaapi.c
index 95a68e62c5..07aa12d403 100644
--- a/libavutil/hwcontext_vaapi.c
+++ b/libavutil/hwcontext_vaapi.c
@@ -1631,8 +1631,10 @@ static void vaapi_device_free(AVHWDeviceContext *ctx)
         XCloseDisplay(priv->x11_display);
 #endif
 
+#if HAVE_VAAPI_DRM
     if (priv->drm_fd >= 0)
         close(priv->drm_fd);
+#endif
 
     av_freep(&priv);
 }
-- 
2.39.5 (Apple Git-154)

