From c2c95a717a90ef7c07c9fd9b773063b722eb7cec Mon Sep 17 00:00:00 2001
From: wang-bin <wbsecg1@gmail.com>
Date: Mon, 13 Nov 2017 12:26:26 +0800
Subject: [PATCH 09/22] videotoolbox: add p010

---
 libavutil/hwcontext_videotoolbox.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/libavutil/hwcontext_videotoolbox.c b/libavutil/hwcontext_videotoolbox.c
index 6eac2c0774..6e464b5ca9 100644
--- a/libavutil/hwcontext_videotoolbox.c
+++ b/libavutil/hwcontext_videotoolbox.c
@@ -36,6 +36,7 @@ static const struct {
     uint32_t cv_fmt;
     enum AVPixelFormat pix_fmt;
 } cv_pix_fmts[] = {
+    { kCVPixelFormatType_420YpCbCr10BiPlanarVideoRange, AV_PIX_FMT_P010 },
     { kCVPixelFormatType_420YpCbCr8Planar,              AV_PIX_FMT_YUV420P },
     { kCVPixelFormatType_422YpCbCr8,                    AV_PIX_FMT_UYVY422 },
     { kCVPixelFormatType_32BGRA,                        AV_PIX_FMT_BGRA },
-- 
2.20.1 (Apple Git-117)

