From 4746cfc06082dbd6a2ba9379e7416799a9ae3ff9 Mon Sep 17 00:00:00 2001
From: wang-bin <wbsecg1@gmail.com>
Date: Tue, 11 Sep 2018 23:16:49 +0800
Subject: [PATCH 13/29] configure: enable icf

---
 configure | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/configure b/configure
index c33d93a7d1..240b8b438e 100755
--- a/configure
+++ b/configure
@@ -7531,6 +7531,9 @@ test_cc -mno-red-zone <<EOF && noredzone_flags="-mno-red-zone"
 int x;
 EOF
 
+check_ldflags --icf=safe || check_ldflags -Wl,--icf=safe || { # ld or clang7 + lld
+    check_cflags -faddrsig && (check_ldflags --icf=all || check_ldflags -Wl,--icf=all)
+}
 
 if enabled icc; then
     # Just warnings, no remarks
-- 
2.39.3 (Apple Git-146)

