From cc32ccae64ee14da6407da2645063673319e87cc Mon Sep 17 00:00:00 2001
From: wang-bin <wbsecg1@gmail.com>
Date: Wed, 28 Feb 2018 14:17:11 +0800
Subject: [PATCH 13/20] check dn arm directive & fix swscale build

clang does not support .dn
---
 configure | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/configure b/configure
index 639021dbc6..857e485d71 100755
--- a/configure
+++ b/configure
@@ -2232,6 +2232,7 @@ TOOLCHAIN_FEATURES="
     as_arch_directive
     as_dn_directive
     as_fpu_directive
+    as_dn_directive
     as_func
     as_object_arch
     asm_mod_q
@@ -5700,6 +5701,7 @@ EOF
     check_as as_arch_directive ".arch armv7-a"
     check_as as_dn_directive   "ra .dn d0.i16"
     check_as as_fpu_directive  ".fpu neon"
+    check_as as_dn_directive   "x .dn 0"
 
     # llvm's integrated assembler supports .object_arch from llvm 3.5
     [ "$objformat" = elf32 ] || [ "$objformat" = elf64 ] &&
-- 
2.11.0

