From 7cb541e63712c9389521850c57bf042bd9583ecc Mon Sep 17 00:00:00 2001
From: wang-bin <wbsecg1@gmail.com>
Date: Fri, 30 Mar 2018 10:46:43 +0800
Subject: [PATCH 15/21] configure: do not filter -libpath for msvc

---
 configure | 1 +
 1 file changed, 1 insertion(+)

diff --git a/configure b/configure
index 8e8bdf032d..4bdd6b62c9 100755
--- a/configure
+++ b/configure
@@ -4329,6 +4329,7 @@ msvc_common_flags(){
             -lz)                  echo zlib.lib ;;
             -lx264)               echo libx264.lib ;;
             -lstdc++)             ;;
+            -libpath:*|-LIBPATH:*)echo $flag ;;
             -l*)                  echo ${flag#-l}.lib ;;
             -LARGEADDRESSAWARE)   echo $flag ;;
             -L*)                  echo -libpath:${flag#-L} ;;
-- 
2.20.1 (Apple Git-117)

