From a8ad6fb4f1cb991bc6cdbb095ab666c7eecf9610 Mon Sep 17 00:00:00 2001
From: wang-bin <wbsecg1@gmail.com>
Date: Mon, 24 Dec 2018 16:06:30 +0800
Subject: [PATCH 15/17] clang-cl: fix undeclared _byteswap_ulong

---
 libavutil/x86/bswap.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/libavutil/x86/bswap.h b/libavutil/x86/bswap.h
index ffa59e4c82..c226cdf62f 100644
--- a/libavutil/x86/bswap.h
+++ b/libavutil/x86/bswap.h
@@ -27,6 +27,7 @@
 #include <stdint.h>
 #if defined(_MSC_VER)
 #include <intrin.h>
+#include <stdlib.h> // clang cl
 #endif
 #include "config.h"
 #include "libavutil/attributes.h"
-- 
2.20.1 (Apple Git-117)

