From e19438ab0b2bc367e99d1787b41c67260cbf6d88 Mon Sep 17 00:00:00 2001
From: wang-bin <wbsecg1@gmail.com>
Date: Mon, 24 Dec 2018 16:06:30 +0800
Subject: [PATCH 15/20] clang-cl: fix undeclared _byteswap_ulong

---
 libavutil/x86/bswap.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/libavutil/x86/bswap.h b/libavutil/x86/bswap.h
index b2f18b6c93..99273e69b2 100644
--- a/libavutil/x86/bswap.h
+++ b/libavutil/x86/bswap.h
@@ -28,6 +28,7 @@
 #if defined(_MSC_VER)
 #include <stdlib.h>
 #include <intrin.h>
+#include <stdlib.h> // clang cl
 #endif
 #include "config.h"
 #include "libavutil/attributes.h"
-- 
2.24.3 (Apple Git-128)

