From c050e1031d2d08591af0dc188ba594d4139ede87 Mon Sep 17 00:00:00 2001
From: wang-bin <wbsecg1@gmail.com>
Date: Sun, 10 Nov 2019 19:01:27 +0800
Subject: [PATCH 18/18] qsvenc: conditionally enable vp9 code

---
 libavcodec/qsvenc.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/libavcodec/qsvenc.c b/libavcodec/qsvenc.c
index 93d49ba21d..1bc5600061 100644
--- a/libavcodec/qsvenc.c
+++ b/libavcodec/qsvenc.c
@@ -824,7 +824,9 @@ static int qsv_retrieve_enc_vp9_params(AVCodecContext *avctx, QSVEncContext *q)
 #endif
 
     mfxExtBuffer *ext_buffers[] = {
+#if QSV_HAVE_EXT_VP9_PARAM
         (mfxExtBuffer*)&vp9_extend_buf,
+#endif
 #if QSV_HAVE_CO2
         (mfxExtBuffer*)&co2,
 #endif
-- 
2.21.0 (Apple Git-122.2)

