From a2c62aa347a1206a8dfb7a08db28802ae9afcbda Mon Sep 17 00:00:00 2001
From: wang-bin <wbsecg1@gmail.com>
Date: Wed, 7 Feb 2024 00:21:02 +0800
Subject: [PATCH 31/31] msvc: skip vvc x86 code for arm64, fix link error

---
 libavcodec/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libavcodec/Makefile b/libavcodec/Makefile
index 470d7cb9b1..e705227354 100644
--- a/libavcodec/Makefile
+++ b/libavcodec/Makefile
@@ -65,7 +65,7 @@ OBJS = ac3_parser.o                                                     \
 
 # subsystems
 include $(SRC_PATH)/libavcodec/vvc/Makefile
-include $(SRC_PATH)/libavcodec/x86/vvc/Makefile
+-include $(SRC_PATH)/libavcodec/$(ARCH)/vvc/Makefile
 OBJS-$(CONFIG_AANDCTTABLES)            += aandcttab.o
 OBJS-$(CONFIG_AC3DSP)                  += ac3dsp.o ac3.o ac3tab.o
 OBJS-$(CONFIG_ADTS_HEADER)             += adts_header.o mpeg4audio_sample_rates.o
-- 
2.39.3 (Apple Git-145)

