From 75591f9cf66add80d3a5f0a060d92fe8813df1cd Mon Sep 17 00:00:00 2001
From: wang-bin <wbsecg1@gmail.com>
Date: Fri, 30 Mar 2018 10:46:43 +0800
Subject: [PATCH 12/33] configure: do not filter -libpath for msvc

---
 configure | 1 +
 1 file changed, 1 insertion(+)

diff --git a/configure b/configure
index 22d8f5c7a3..51fc4f3038 100755
--- a/configure
+++ b/configure
@@ -4945,6 +4945,7 @@ msvc_common_flags(){
             -lz)                  echo zlib.lib ;;
             -lx264)               echo libx264.lib ;;
             -lstdc++)             ;;
+            -libpath:*|-LIBPATH:*)echo $flag ;;
             -l*)                  echo ${flag#-l}.lib ;;
             -LARGEADDRESSAWARE)   echo $flag ;;
             -L*) [ "$_flags_type" = "link" ] && echo -libpath:${flag#-L} ;;
-- 
2.49.0

