From 2ea8c56cf573c2990f7f5acf54aaff7fa6d6e057 Mon Sep 17 00:00:00 2001
From: wang-bin <wbsecg1@gmail.com>
Date: Fri, 30 Mar 2018 10:46:43 +0800
Subject: [PATCH 12/36] configure: do not filter -libpath for msvc

---
 configure | 1 +
 1 file changed, 1 insertion(+)

diff --git a/configure b/configure
index be1447aafb..511bf0bd11 100755
--- a/configure
+++ b/configure
@@ -5212,6 +5212,7 @@ msvc_common_flags(){
             -lz)                  echo zlib.lib ;;
             -lx264)               echo libx264.lib ;;
             -lstdc++)             ;;
+            -libpath:*|-LIBPATH:*)echo $flag ;;
             -l*)                  echo ${flag#-l}.lib ;;
             -LARGEADDRESSAWARE)   echo $flag ;;
             -L*) [ "$_flags_type" != "link" ] || echo -libpath:${flag#-L} ;;
-- 
2.53.0

