From be2f0abfe9a7b195b01436b3a231f804adfef0ce Mon Sep 17 00:00:00 2001
From: wang-bin <wbsecg1@gmail.com>
Date: Sun, 14 Apr 2019 11:42:42 +0800
Subject: [PATCH 14/35] configure: suppor static build via clang-cl

---
 configure | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/configure b/configure
index 5834943328..fe5447bc3f 100755
--- a/configure
+++ b/configure
@@ -6223,7 +6223,15 @@ case $target_os in
             SHFLAGS='-dll -def:$$(@:$(SLIBSUF)=.def) -implib:$(SUBDIR)$(SLIBNAME:$(SLIBSUF)=.lib)'
         fi
         enabled x86_64 && objformat="win64" || objformat="win32"
-        ranlib=:
+        # ranlib is required by llvm toolchain, clang-cl is detected as msvc
+        [ "$toolchain" = msvc ] && {
+          ranlib=:
+        } || {
+          LIBPREF= # patch win clang static lib
+          LIBSUF=.lib
+          LD_LIB='%.lib'
+          SLIB_CREATE_DEF_CMD='EXTERN_PREFIX="$(EXTERN_PREFIX)" AR="$(AR_CMD)" NM="$(NM_CMD)" $(SRC_PATH)/compat/windows/makedef $(SUBDIR)lib$(NAME).ver $(OBJS) > $$(@:$(SLIBSUF)=.def)'
+        }
         enable dos_paths
         ;;
     cygwin*)
-- 
2.54.0 (Apple Git-157)

