From f73d42c0f75567e3ff8ebe7482df2276f64e37d4 Mon Sep 17 00:00:00 2001
From: wang-bin <wbsecg1@gmail.com>
Date: Fri, 3 Jul 2020 14:09:54 +0800
Subject: [PATCH 17/33] nvenc: check sdk 10.0 ptrs at runtime

---
 libavcodec/nvenc.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/libavcodec/nvenc.c b/libavcodec/nvenc.c
index 08cdcbae02..2ec9787d51 100644
--- a/libavcodec/nvenc.c
+++ b/libavcodec/nvenc.c
@@ -1665,17 +1665,18 @@ static av_cold int nvenc_setup_encoder(AVCodecContext *avctx)
     else if (ctx->flags & NVENC_LOWLATENCY)
         ctx->init_encode_params.tuningInfo = NV_ENC_TUNING_INFO_LOW_LATENCY;
 
+    if (p_nvenc->nvEncGetEncodePresetConfigEx)
     nv_status = p_nvenc->nvEncGetEncodePresetConfigEx(ctx->nvencoder,
         ctx->init_encode_params.encodeGUID,
         ctx->init_encode_params.presetGUID,
         ctx->init_encode_params.tuningInfo,
         &preset_config);
-#else
+    else
+#endif
     nv_status = p_nvenc->nvEncGetEncodePresetConfig(ctx->nvencoder,
         ctx->init_encode_params.encodeGUID,
         ctx->init_encode_params.presetGUID,
         &preset_config);
-#endif
     if (nv_status != NV_ENC_SUCCESS)
         return nvenc_print_error(avctx, nv_status, "Cannot get the preset configuration");
 
-- 
2.39.5 (Apple Git-154)

