From 600e7cbb1fefc1841963df7697a8111639d98586 Mon Sep 17 00:00:00 2001
From: wang-bin <wbsecg1@gmail.com>
Date: Sun, 13 Mar 2022 11:15:15 +0800
Subject: [PATCH 24/30] win: enable windres for small and msvc

---
 configure | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/configure b/configure
index dc8b91c788..cb27834125 100755
--- a/configure
+++ b/configure
@@ -5592,7 +5592,7 @@ case $target_os in
             # Cannot build both shared and static libs when using dllimport.
             disable static
         fi
-        ! enabled small && test_cmd $windres --version && enable gnu_windres
+        test_cmd $windres --version && enable gnu_windres
         enabled x86_32 && check_ldflags -Wl,--large-address-aware
         shlibdir_default="$bindir_default"
         SLIBPREF=""
@@ -5642,7 +5642,7 @@ case $target_os in
             # Cannot build both shared and static libs with MSVC or icl.
             disable static
         fi
-        ! enabled small && test_cmd $windres --version && enable gnu_windres
+        test_cmd $windres --version && enable gnu_windres
         enabled x86_32 && check_ldflags -LARGEADDRESSAWARE
         shlibdir_default="$bindir_default"
         SLIBPREF=""
-- 
2.39.3 (Apple Git-145)

